src/PeopleList.tsx

```diff
@@ -0,0 +1,91 @@
+import { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+import classNames from 'classnames';
+
+import { Person } from './types/Person';
+
+interface Props {
+  people: Person[];
+  onSelect: (persone: Person) => void;
+  delay: number;
+}
+
+export const Peoplelist: React.FC<Props> = ({ people, onSelect, delay }) => {
```

Comment:
In the future try to choose more descriptive component names.
When I see the PeopleList name I (as a developer who's seeing your code at the first time) expect to find only some map of the list in it. But in your case it containes some input and logic for applying query.

Reply:
Ideally it should be some component for instance - Autocomplete that contains input + logic and another component - people list which only accepts currentPeople as a prop and renders it.

Reply:
no need to refactor it now. Just FYI)